server/routes/profile.js

```diff
@@ -0,0 +1,15 @@
+const express = require("express");
+const router = express.Router();
+const protect = require("../middleware/auth");
+const {
+  profileCreatePost,
+  profileUpdatePost,
+  profileGet,
+  profileGetAll,
+} = require("../controllers/profile");
+router.route("/create").post(protect, profileCreatePost);
+router.route("/update").put(protect, profileUpdatePost);
```

Comment:
```suggestion
router.route("/").put(protect, profileUpdatePost);
```

Reply:
Yes, the protect middleware decodes the jwt token which contains the user's id, which is then added to the request object